grisafi/app.js

```diff
@@ -0,0 +1,265 @@
+var express= require('express');
+var app = express();
+var bodyParser= require('body-parser');
+app.use(bodyParser.json());
+
+const DIA_EN_MILISEGUNDOS = 1000*60*60*24;
+
+var libros = new Array();
+var socios = new Array();
+var prestamos = new Array();
+
+function Libro(titulo,cantidad,id){
+    this.id=id;
+    this.titulo=titulo;
+    this.cantidad=cantidad;
+    this.disponibles=function(){
+        let count=0;
+        for (const prest of prestamos) {
+            if (prest.idLibro==this.id){
+                count++;
+            }
+        }
+        return this.cantidad-count;
+    }
+}
+
+function Socio(nombre,id){
+    this.id=id;
+    this.nombre=nombre;
+}
+
+function Prestamo(id,idSocio,idLibro,dias){
+    this.id=id;
+    this.idSocio=idSocio;
+    this.idLibro=idLibro;
+    this.fechavencimiento=Date.now()+DIA_EN_MILISEGUNDOS*dias;
+}
+
+function librosPrestados(idLibro){
+    let cont=0;
+    for (let i = 0; i < prestamos.length; i++) {
+        if(prestamos[i].idLibro==idLibro){
+            cont++;
+        }
+    }
+    return cont;
+}
+
+function updateBook(bookID,cantidad){
+    if(cantidad>librosPrestados(bookID)){
+        for (let i = 0; i < libros.length; i++) {
+            if(bookID == libros[i].id){
+                libros[i].cantidad=cantidad;
+                return 1;
+            }
+        }
+        return 0;
+    }
+    else{
+        return -1;
+    }
+}
+
+function findID(id,coleccion){
```

Comment:
return a boolean instead of an integer

Reply:
i guess this isn't the best REST API ever made

Reply:
jajaja, it can be!